e2e_tests/tests/command/test_strict_ntsc.py

```diff
@@ -0,0 +1,261 @@
+from typing import Any, List
+
+import pytest
+
+from determined.common import api
+from determined.common.api import authentication, bindings
+from determined.common.api.errors import APIException, BadRequestException, NotFoundException
+from tests import command as cmd
+from tests import config as conf
+from tests import experiment as exp
+from tests.cluster.test_users import ADMIN_CREDENTIALS, create_test_user, logged_in_user
+
+
+def assert_shell_access(creds: authentication.Credentials, shell_id: str, can_access: bool) -> None:
+    master_url = conf.make_master_url()
+    authentication.cli_auth = authentication.Authentication(
+        master_url, creds.username, creds.password, try_reauth=True
+    )
+    sess = api.Session(master_url, None, None, None)
+
+    resp = bindings.get_GetShells(sess)
+    shell_ids = [shell.id for shell in resp.shells or []]
+    if can_access:
+        assert shell_id in shell_id
+        bindings.get_GetShell(sess, shellId=shell_id)
+        req = bindings.v1SetShellPriorityRequest(shellId=shell_id, priority=50)
+        bindings.post_SetShellPriority(sess, shellId=shell_id, body=req)
+        api.get(master_url, f"shells/{shell_id}/events", auth=authentication.cli_auth)
+        with api.ws(master_url, f"shells/{shell_id}/events") as ws:
+            for _ in ws:
+                break
+        return
+
+    assert shell_id not in shell_ids
+    with pytest.raises(NotFoundException):
+        bindings.get_GetShell(sess, shellId=shell_id)
+    with pytest.raises(NotFoundException):
+        bindings.post_KillShell(sess, shellId=shell_id)
+    with pytest.raises(NotFoundException):
+        req = bindings.v1SetShellPriorityRequest(shellId=shell_id, priority=50)
+        bindings.post_SetShellPriority(sess, shellId=shell_id, body=req)
+    with pytest.raises(NotFoundException):
+        api.get(master_url, f"shells/{shell_id}/events", auth=authentication.cli_auth)
```

Comment:
do we test if "does this shell id show up in the list of shells" anywhere?

Reply:
for the negative case we test on line 34  ```assert shell_id not in shell_id``` 

and for the positive case we test on line 24